src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersFixer.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>Fixer for <see cref="UseExceptionThrowHelpers"/>.</summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseExceptionThrowHelpersFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            UseExceptionThrowHelpers.UseArgumentNullExceptionThrowIfNullRuleId,
+            UseExceptionThrowHelpers.UseArgumentExceptionThrowIfNullOrEmptyRuleId,
+            UseExceptionThrowHelpers.UseArgumentOutOfRangeExceptionThrowIfRuleId,
+            UseExceptionThrowHelpers.UseObjectDisposedExceptionThrowIfRuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => CustomFixAllProvider.Instance;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            SemanticModel model = await doc.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode root = await doc.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (TryGetFixInfo(doc, root, model, context.Diagnostics[0], out INamedTypeSymbol? typeSymbol, out string? methodName, out SyntaxNode? node, out SyntaxNode? arg, out SyntaxNode? other))
+            {
+                string title = string.Format(MicrosoftNetCoreAnalyzersResources.UseThrowHelperFix, typeSymbol.Name, methodName);
```

Comment:
NIT: I had a comment that separate resource messages are preferred for localization purpose

Reply:
What would be different about each of them?

Reply:
Oh right, the exception and method name will not be translated